tests/suite/test_transport_server_status.py

```diff
@@ -0,0 +1,114 @@
+import pytest
+
+from suite.resources_utils import wait_before_test
+from suite.custom_resources_utils import (
+    read_v1alpha1_crd,
+    patch_transport_server_from_yaml,
+)
+from settings import TEST_DATA
+
+
+@pytest.mark.ts
+@pytest.mark.parametrize(
+    "crd_ingress_controller, transport_server_setup",
+    [
+        (
+            {
+                "type": "complete",
+                "extra_args":
+                    [
+                        "-enable-custom-resources",
+                        "-global-configuration=nginx-ingress/nginx-configuration"
+                    ]
+            },
+            {"example": "transport-server-status", "app_type": "simple"},
+        )
+    ],
+    indirect=True,
+)
+class TestTransportServerStatus:
+
+    def restore_transport_server(self, kube_apis, transport_server_setup) -> None:
+        """
+        Function to revert a TransportServer deployment to a valid state.
+        """
+        patch_src = f"{TEST_DATA}/transport-server-status/standard/transport-server.yaml"
+        patch_transport_server_from_yaml(
+            kube_apis.custom_objects,
+            transport_server_setup.name,
+            patch_src,
+            transport_server_setup.namespace,
+        )
+
+    @pytest.mark.smoke
+    def test_status_valid(
+        self, kube_apis, crd_ingress_controller, transport_server_setup,
+    ):
+        """
+        Test TransportServer status with valid fields in yaml.
+        """
+        response = read_v1alpha1_crd(
+            kube_apis.custom_objects,
+            transport_server_setup.namespace,
+            "transportservers",
+            transport_server_setup.name,
+        )
+        assert (
+            response["status"]
+            and response["status"]["reason"] == "AddedOrUpdated"
+            and response["status"]["state"] == "Valid"
+        )
+
+    def test_status_warning(
+        self, kube_apis, crd_ingress_controller, transport_server_setup,
+    ):
+        """
+        Test TransportServer status with a missing listener.
+        """
+        patch_src = f"{TEST_DATA}/transport-server-status/rejected-warning.yaml"
+        patch_transport_server_from_yaml(
+            kube_apis.custom_objects,
+            transport_server_setup.name,
+            patch_src,
+            transport_server_setup.namespace,
+        )
+        wait_before_test()
+        response = read_v1alpha1_crd(
```

Comment:
we have a number functions specific for TransportServers -- `patch_transport_server_from_yaml`, `create_ts`, `delete_ts`. However, `read_v1alpha1_crd` is not TransportServer specific. 

Would it make sense to also have `read_ts`?

Reply:
I've updated the method names, wrapping a read_custom_resource_v1alpha.